llpc/lower/llpcSpirvLowerRayQuery.cpp

```diff
@@ -1282,28 +1282,21 @@ bool SpirvLowerRayQuery::stageNotSupportLds(ShaderStage stage) {
 }
 
 // =====================================================================================================================
-// Create instructions to load instance index given the 64-bit instance node address at the current insert point
+// Create instructions to load instance index/id given the 64-bit instance node address at the current insert point
 //
 // @param instNodeAddr : 64-bit instance node address, in <2 x i32>
-Value *SpirvLowerRayQuery::createLoadInstanceIndex(Value *instNodeAddr) {
-  Value *zero = m_builder->getInt32(0);
-  Type *gpuAddrAsPtrTy = PointerType::get(*m_context, SPIRAS_Global);
-  auto int32x2Ty = FixedVectorType::get(m_builder->getInt32Ty(), 2);
-
-  const unsigned instanceIndexOffset = offsetof(RayTracingInstanceNode, extra.instanceIndex);
-
-  Value *instanceIndexOffsetVar = PoisonValue::get(int32x2Ty);
-  instanceIndexOffsetVar =
-      m_builder->CreateInsertElement(instanceIndexOffsetVar, m_builder->getInt32(instanceIndexOffset), uint64_t(0));
-  instanceIndexOffsetVar = m_builder->CreateInsertElement(instanceIndexOffsetVar, zero, 1);
-  Value *instanceIndexAddr = m_builder->CreateAdd(instNodeAddr, instanceIndexOffsetVar);
-
-  instanceIndexAddr = m_builder->CreateBitCast(instanceIndexAddr, m_builder->getInt64Ty());
-  auto instanceIndexAddrAsPtr = m_builder->CreateIntToPtr(instanceIndexAddr, gpuAddrAsPtrTy);
-  auto loadValue = m_builder->CreateConstGEP1_32(m_builder->getInt8Ty(), instanceIndexAddrAsPtr, 0);
-  loadValue = m_builder->CreateBitCast(loadValue, PointerType::get(*m_context, SPIRAS_Global));
-
-  return m_builder->CreateLoad(m_builder->getInt32Ty(), loadValue);
+Value *SpirvLowerRayQuery::createLoadInstanceIndexOrId(Value *instNodeAddr, bool isIndex) {
+  auto instanceIdAddr = m_builder->CreateBitCast(instNodeAddr, m_builder->getInt64Ty());
+  Value *instanceIdPtr = m_builder->CreateAlloca(m_builder->getInt64Ty());
+  m_builder->CreateStore(instanceIdAddr, instanceIdPtr);
+
+  StringRef rayQueryGetInstanceIndx =
```

Comment:
How about just putting the attribute value in a separate variable and call getRaytracingFunctionName() with the variable name in the call to CreateNamedCall below?

Reply:
> How about just putting the attribute value in a separate variable and call getRaytracingFunctionName() with the variable name in the call to CreateNamedCall below?

yes, we can call the lambda func in startswith() directly.